ui-react/packages/atlasmap-core/src/models/expression.model.ts

```diff
@@ -73,7 +73,7 @@ export class FieldNode extends ExpressionNode {
         this.field = mapping.getMappedFieldByName(fieldParts[1], true)!;
       } else {
         // TODO: check this non null operator
-        this.field = mapping.getMappedFieldForIndex((index + 1).toString(), true)!;
+        this.field = mapping.getMappedFieldForIndex((index).toString(), true)!;
```

Comment:
Doesn't this cause a problem for the index editor in the mapping detail pane? They are 1-base.

Reply:
No - see mapping.model change below - it immediately subtracts the 1 from the index.  This was a real index arg - not a user index argf.

Reply:
OK, so index editor only uses `getIndexForMappedField()` here
https://github.com/atlasmap/atlasmap/blob/master/ui/src/app/lib/atlasmap-data-mapper/components/mapping-detail/mapping-field-detail.component.html#L46

Then if we change the `getMappedFieldForIndex()` to be 0-base, we also have to change `getIndexForMappedField()` to be 0-base to avoid mixing 0-base and 1-base. In that case `MappingFieldDetail` component have to do +1 by itself to be 1-base.

Reply:
Went back to the original index formula